src/js/controller/controller.js

```diff
@@ -486,6 +486,7 @@ Object.assign(Controller.prototype, {
         function _pause(meta) {
             _actionOnAttach = null;
             checkAutoStartCancelable.cancel();
+            checkAutoStartCancelable = cancelable(_checkAutoStart);
```

Comment:
@vseventer I notice we're also doing this on `_load()`, but not not when cancelling on `_play()` and `_stop()`.

The reason we reset the cancelable on load is because loading a new playlist results in an autostart check. play, stop and even pause should cancel autostart behavior.

If you pause while the player is attempting to start, it shouldn't start, or at least it should start and then pause. I'm not sure that's what will happen with this change.

Reply:
@robwalch would it suffice to check the pause reason here to see if it's from a viewability change and not a user interaction or api call to `pause()`? This may be simpler than changing `_checkPlayOnViewable`.
```
if (pauseReason === 'autostart') {
    checkAutoStartCancelable = cancelable(_checkAutoStart);
}
```

Reply:
I don't think so. We should look at the timing of event listeners. There is not reason to pause on "viewable" until playback has started, so splitting up the "viewable" event handler to one that autostarts, and another that pauses once playback has started is the right move.